### ISSUE TITLE
Custom rules in multiple modules project

### ISSUE THREAD

**Issue by varunmendiratta** [detekt/detekt#3989]
## Expected Behavior
Custom rule violations should be reported for a project which has multiple modules and a custom Gradle task.

## Observed Behavior
When `detekt` is run on terminal while specifying 'custom rules project JAR', it reports custom rules violations. But it does not report violations when custom Gradle task is run

## Steps to Reproduce
For adding custom rules, I have created another module as mentioned [here](https://github.com/detekt/detekt/issues/3897) `customdetekt`.

Here is the root level build.gradle. I have a custom Gradle task `detektAll `
```
import io.gitlab.arturbosch.detekt.Detekt

plugins {
    id "io.gitlab.arturbosch.detekt" version "1.17.1"
}

tasks.register("detektAll", Detekt) {
    dependsOn(":customdetekt:assemble")
    description = "Custom Detekt build for all modules"
    parallel = true
    setSource(file(projectDir))
    config.setFrom(files("$rootDir/detekt.yml"))
    jvmTarget = "1.8"
    classpath.setFrom(project.configurations.getByName("detekt"))
    include("**/*.kt")
    exclude("**/build/**")
    reports {
        html {
            enabled = true
            destination = file("$rootDir/build/reports/detekt-results.html")
        }
        xml {
            enabled = true
            destination = file("$rootDir/build/reports/detekt-results.xml")
        }
        txt.enabled = false
    }
}
```

Here is the app level `build.gradle`
```
apply plugin: 'io.gitlab.arturbosch.detekt'
dependencies {
	detektPlugins project(':customdetekt')
}
```

Here is the `build.gradle` file of `customdetekt`
```
apply plugin: "kotlin"

repositories {
    mavenCentral()
}

dependencies {
    compileOnly "io.gitlab.arturbosch.detekt:detekt-api:1.17.1"
    compileOnly "org.jetbrains.kotlin:kotlin-stdlib:1.5.21"
}
```

I realised that `customdetekt.jar` file is created & when I call `detekt` from terminal specifying this JAR, it works fine and shows all violations of my custom rules.
`detekt -c detekt.yml -p customdetekt/build/libs/customdetekt.jar -r html:detekt-report.html`

But it does not show custom rule violations when I run `./gradlew detektAll`
What I am doing wrong? Did I miss any step?

## Context
I want to create custom rules on my project which has multiple modules

## Your Environment

* Version of detekt used: 1.17.1
* Version of Gradle used (if applicable): 4.2.1
* Gradle scan link (add `--scan` option when running the gradle task):
* Operating System and version: MacOS 10.15.6
* Link to your project (if it's a public repository):

**Comment by cortinico** [comment]
> What I am doing wrong? Did I miss any step?

The problem here is that you're defining the `detektPlugins` dependency on the app project, and registering a task `detektAll` on the root project. That won't work as the `detektPlugins` dependency would work only for tasks in the `:app` project.

If you move the `dependency` block on the root project it should work correctly.

**Comment by varunmendiratta** [comment]
The problem is solved now after following @cortinico suggestion.

**Comment by odravison** [comment]
I wish my life's problems be easy to solve like that! xD
